### PR TITLE
[feat] Allow purging modules before running regression tests

### PR DESCRIPTION
**Pull request by trolleyman**
This option allows the user to specify whether or not they want to add a 'module purge' to the start of every job, before each module is loaded.

For example, given a sbatch job like so:
```bash
#!/bin/bash
#SBATCH -t 1:0
module load python
python test.py
```

The sbatch script will be output like this instead:

```bash
#!/bin/bash
#SBATCH -t 1:0
module purge
module load python
python test.py
```

The setting is per-system, so it fits next to the `module_system` option.

### CODE REVIEW

**Comment by trolleyman** [comment]
@vkarak Ah ok, so should I add a command line option, (`--purge` or something), and then purge in `frontend/cli.py`? Is purging just before the options are logged ([L333](https://github.com/eth-cscs/reframe/blob/master/reframe/frontend/cli.py#L333)) a good idea, or somewhere else?

**Comment by vkarak** [comment]
@trolleyman I'd suggest a `--purge-env` option within the `misc_options` group. Then the right place to do the purge is where we do some similar environment preparations ([L415](https://github.com/eth-cscs/reframe/blob/master/reframe/frontend/cli.py#L415)), just after unloading the user's reframe module (if specified) and before loading any custom user modules specified with the `-m` option. To purge all modules, you may use the [unload_all()](https://eth-cscs.github.io/reframe/reference.html#reframe.core.modules.ModulesSystem.unload_all) method of the modules system.

**Comment by vkarak** [comment]
@jenkins-cscs retry all

File: reframe/frontend/cli.py [hunk 1]
```diff
@@ -177,6 +177,9 @@ def main():
         '--module-mappings', action='store', metavar='FILE',
         dest='module_map_file',
         help='Apply module mappings defined in FILE')
+    misc_options.add_argument(
+        '--purge-env', action='store_true', dest='purge_env', default=False,
+        help='Perform a `module purge\' before setup')
```
[the remark this answers is not in the document]
Comment (vkarak) [reply reframe/frontend/cli.py]: I'd also suggest a slightly more generic message here: "Purge modules environment before running any tests".

**Comment by trolleyman** [comment]
@vkarak Fixed - cheers.

**Pull request merged**